Stardaze/OutputFormatter.swift

```diff
@@ -17,6 +23,119 @@ internal struct PrettyPrintedStringFormatter: Visitor {
         return finishedString
     }
 
+    private func makeCompactDocumentParameters(document: Document) -> [String: Any] {
+        var parameters: [String: Any] = ["query":
+            makeCondensedQuery(document: document, encoded: outputOption == .encoded)]
+
+        if let operationName = document.queryOperation.name {
+            parameters["operationName"] = operationName
+        }
+
+        if let variables = makeCondensedVariables(document: document, encoded: outputOption == .encoded) {
+            parameters["variables"] = variables
+        }
+
+        return parameters
+    }
+
+    private func makeCompactDocumentString(document: Document) -> String {
+        let queryComponent = "query=\(makeCondensedQuery(document: document, encoded: outputOption == .encoded))"
+
+        let operationNameComponent: String = {
+            if let operationName = document.queryOperation.name {
+                return "&operationName=\(operationName)"
+            } else {
+                return ""
+            }
+        }()
+
+        let variablesComponent: String = {
+            if let variables = makeCondensedVariables(document: document, encoded: outputOption == .encoded) {
+                return "&variables=\(variables)"
+            } else {
+                return ""
+            }
+        }()
+
+        return "\(queryComponent)\(operationNameComponent)\(variablesComponent)"
+    }
+
+    private func makeCondensedVariables(document: Document, encoded: Bool) -> String? {
+        if let variablesDefinitionList = document.queryOperation.variableDefinitions {
+            let variables = makeReadableVariableValueListString(variableDefinitionList:
+                variablesDefinitionList).condensingWhitespace()
+
+            if encoded {
+                if let encodedVariables = variables.addingPercentEncoding(withAllowedCharacters:
+                    CharacterSet.urlQueryAllowed) {
+                    return encodedVariables
+                } else {
+                    return nil
+                }
+            } else {
+                return variables
+            }
+        } else {
+            return nil
+        }
```

Comment:
I would recommend refactoring this method to use `guard` statements for unwrapping optionals, as that would prevent all of the nesting that is going on here.